### PR TITLE
LEAF-2601 Delete Step from workflow

### PR DESCRIPTION
**Pull request by jrodcode**
Ticket 2601 - This will help delete the initial step from a workflow, and detach it from the records_workflow_state.

### CODE REVIEW

File: LEAF_Request_Portal/sources/Workflow.php [hunk 1]
```diff
@@ -71,10 +71,8 @@ public function getSteps()
         return $out;
     }
 
-    public function deleteStep($stepID)
-    {
-        if (!$this->login->checkGroup(1))
-        {
+    public function deleteStep($stepID){
```
Comment (raphaelfretter) [inline LEAF_Request_Portal/sources/Workflow.php:74]:
```suggestion
    public function deleteStep(int $stepID): int|string {
```

**Review by raphaelfretter — CHANGES_REQUESTED** [review]

File: LEAF_Request_Portal/sources/Workflow.php (hunk 1 above)
Comment (jrodcode) [reply LEAF_Request_Portal/sources/Workflow.php:74]: Great addition, I tested it and it works well! thank you.

**Review by shaneodd — APPROVED** [review]

**Review by raphaelfretter — APPROVED** [review]

**Review by AmandaSumner — APPROVED** [review]

**Review by aerinkayne — COMMENTED** [review]
From the Jira description, it looks like workflow step deletion should only succeed when there are records at the step if all those records are in a deleted state.  My understanding is that if there are any active requests at the step that those would still need to be manually deleted by initiators before the step can be removed.

**Review by Pelentan — APPROVED** [review]

**Review by aerinkayne — CHANGES_REQUESTED** [review]
We should discuss some questions about this in more detail

**Pull request closed without merging**